napari/plugins/_tests/test_utils.py

```diff
@@ -198,6 +198,15 @@ def read_all(path):
     assert len(readers) == 2
 
 
+def test_get_potential_readers_extension_case(tmp_plugin: DynamicPlugin):
+    @tmp_plugin.contribute.reader(filename_patterns=['*.tif'])
+    def read_tif(path):
+        ...
+
+    readers = get_potential_readers('my_file.TIF')
+    assert len(readers) == 2
```

Comment:
Why 2?

Reply:
builtin?

Reply:
I copied another test and forgot to change, whoops :sweat_smile:

Reply:
But I haven't fixed the bug anyway:
```python
      def test_get_potential_readers_extension_case(tmp_plugin: DynamicPlugin):
          @tmp_plugin.contribute.reader(filename_patterns=['*.tif'])
          def read_tif(path):
              ...
      
          readers = get_potential_readers('my_file.TIF')
  >       assert len(readers) == 2
  E       assert 0 == 2
  E        +  where 0 = len({})
```

Will investigate further.

Reply:
the core problem of test failure is that you patch the method of QtViewer, not `get_potential_readers`, and in the test, you call directly `get_potential_readers`, so your new code is not used. 

When I copy your code in the `get_potential_readers` body, it works (@brisvag is correct and should find one reader):

```
        readers = get_potential_readers('my_file.TIF')
>       assert len(readers) == 2
E       AssertionError: assert 1 == 2
E        +  where 1 = len({'tmp_plugin': 'tmp_plugin'})
```

Reply:
Yup, I find one reader as well in my testing env (and indeed builtins should not exist at this point as far as I can tell, so 1 is correct).

Reply:
Agreed, as I said I copied and forgot to change the number!